src/widget/wtracktableview.cpp

```diff
@@ -1597,6 +1603,22 @@ void WTrackTableView::slotClearBeats() {
     }
 }
 
+void WTrackTableView::slotClearWaveform() {
+    TrackModel* trackModel = getTrackModel();
+    if (trackModel == NULL) {
+        return;
+    }
+
+    QModelIndexList selectedIndices = selectionModel()->selectedRows();
+    foreach (QModelIndex index, selectedIndices) {
+        TrackPointer pTrack = trackModel->getTrack(index);
+        if (pTrack) {
+            pTrack->setWaveform(WaveformPointer());
```

Comment:
This means, you dereference the Waveform, where the TrackInfo member variable is pointing to. 
This does not effect the references that are already spread though Mixxx via getWaveform(). 

So you need to inform all getWaveform() users, that the waveform hold by the original pointer is now invalid. 

Maybe we can set an atomic flag in the original waveform ..